include/hip/hcc_detail/hip_fp16.h

```diff
@@ -34,7 +34,7 @@ THE SOFTWARE.
     #include <utility>
 #endif
 
-#if defined(__clang__) && (__clang_major__ > 5)
+#if __HCC__ || __HIP__ 
```

Comment:
Should this include check for defined(__clang__) too as done in https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/hcc_detail/hip_common.h#L30?
OR could we use __HCC_OR_HIP_CLANG__ ?

Reply:
replaced with HCC_OR_HIP_CLANG as suggested